---
title: "Quantifying rhythmicity with LimoRhyde2"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{quantifying-rhythmicity}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  rmarkdown.html_vignette.check_title = FALSE
)
```

## Introduction

We demonstrate a basic function of LimoRhyde2, quantifying rhythmicity, on gene expression data from mouse liver tissue samples.

## Setup

We begin by preparing the R environment.

```{r setup}
library(limorhyde2)
library(data.table)
library(ggplot2)
library(qs)
```

## Loading the data

Next, we load the sample data.

```{r}
mouseLiverData = qread(system.file('data', 'mouse_liver_d.qs', package = 'limorhyde2')) 
mouseLiverMetadata = qread(system.file('data', 'mouse_liver_md.qs', package = 'limorhyde2'))
```

## Fitting a limorhyde2 object

The first step of the LimoRhyde2 workflow is to fit a series of linear models for each genomic feature. For the example data, the name of the column containing the time at which a sample was taken is 'time'.

```{r, results='hide'}
fit = getModelFit(y = mouseLiverData, metadata = mouseLiverMetadata, 
                timeColname = 'time')
```

## Getting a posterior fit

Next, we obtain a model with moderated coefficient estimates using `getPosteriorFit`. By default, the posterior fit is obtained by moderating the raw fit based on the patterns of covariance between features (as well as conditions and other covariates, if applicable) present in the data.

```{r, results='hide'}
fit = getPosteriorFit(fit)
```

## Getting rhythmic statistics

From the moderated model, we can get obtain estimates of the rhythmic statistics for a select set of genes using `getRhythmStats` (in our example, we use three clock genes: Per1, Cry2, and Clock). Note that by default, `getRhythmStats` estimates from posterior means of coefficients, but it can also be used with the raw coefficients.

```{r}
clockGeneIds = as.character(c(18626, 12953, 12753))
clockGeneLabs = c('18626' = 'Per1', '12953' = 'Cry2', '12753' = 'Clock') 

rhyStats = getRhythmStats(fit, features = clockGeneIds)
rhyStats
```

## Estimating fitted values 

We can also obtain fits based on the moderated estimates using `getFittedValues`.

```{r, results='hide'}
fittedVals = getFittedValues(fit, times = seq(18, 64, by = 2), features = clockGeneIds)
```

Below, we plot the fitted values from the moderated model. The points show the actual values of the data, while the lines show the fitted values at a given time.

```{r, fig.align='center', fig.width=8, fig.asp=0.5625}
#setting up data table with actual and fitted values for given time
d = mouseLiverData[clockGeneIds,]
d = as.data.table(t(d), keep.rownames = 'sample')
merged = merge(d, mouseLiverMetadata, by = 'sample')
merged = melt(merged, id.vars = c('sample', 'title', 'organ', 'time'), variable.name = 'feature',
     value.name = 'actual_value')
plotDt = merge(merged, fittedVals, by = c('feature', 'time'))
setnames(plotDt, 'value', 'fitted_value')

ggplot(data = plotDt, aes(x = time)) +
  geom_line(aes(y = fitted_value)) +
  geom_point(aes(y = actual_value), color = 'blue') +
  facet_wrap(vars(feature), scales = 'free_y', labeller = labeller(feature = clockGeneLabs)) +
  ylab('Expression') +
  xlab('Sampling time (h)')

```