<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Quantifying rhythmicity in one condition â€¢ limorhyde2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Quantifying rhythmicity in one condition">
<meta property="og:description" content="limorhyde2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">limorhyde2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/differential-rhythmicity.html">Quantifying differential rhythmicity between conditions</a>
    </li>
    <li>
      <a href="../articles/rhythmicity.html">Quantifying rhythmicity in one condition</a>
    </li>
    <li>
      <a href="../articles/rnaseq.html">Analyzing RNA-seq data</a>
    </li>
    <li>
      <a href="../articles/uncertainty.html">Quantifying uncertainty in (differential) rhythmicity</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/hugheylab/limorhyde2/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="rhythmicity_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Quantifying rhythmicity in one condition</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/hugheylab/limorhyde2/blob/HEAD/vignettes/rhythmicity.Rmd" class="external-link"><code>vignettes/rhythmicity.Rmd</code></a></small>
      <div class="hidden name"><code>rhythmicity.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Here we show how to use <code>limorhyde2</code> to quantify rhythmicity in data from one condition. The data are based on mouse liver samples from the <a href="https://doi.org/10.1073/pnas.1408886111" class="external-link">circadian gene expression atlas in mammals</a> (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE54650" class="external-link">GSE54650</a>).</p>
</div>
<div class="section level2">
<h2 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://r-datatable.com" class="external-link">'data.table'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://limorhyde2.hugheylab.org" class="external-link">'limorhyde2'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/traversc/qs" class="external-link">'qs'</a></span><span class="op">)</span>

<span class="co"># doParallel::registerDoParallel() # register a parallel backend to minimize runtime</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-data">Load the data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h2>
<p>The expression data are in a matrix with one row per gene and one column per sample. The metadata are in a table with one row per sample. To save time and space, the expression data include only a subset of genes.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">y</span> <span class="op">=</span> <span class="va">GSE54650</span><span class="op">$</span><span class="va">y</span>
<span class="va">y</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt;           GSM1321182 GSM1321183 GSM1321184 GSM1321185 GSM1321186</span>
<span class="co">#&gt; 13088      10.209968  10.503358   9.714828   9.477728  10.134381</span>
<span class="co">#&gt; 13170       7.871989   7.281691   6.946495   6.266409   7.150140</span>
<span class="co">#&gt; 13869       6.851685   7.406234   7.781294   7.924764   8.089538</span>
<span class="co">#&gt; 100009600   5.055931   5.199767   5.121511   5.155912   5.219873</span>
<span class="co">#&gt; 100418030   6.174130   6.547075   6.521593   6.413817   6.451957</span>

<span class="va">metadata</span> <span class="op">=</span> <span class="va">GSE54650</span><span class="op">$</span><span class="va">metadata</span>
<span class="va">metadata</span>
<span class="co">#&gt;         sample tissue time</span>
<span class="co">#&gt;  1: GSM1321182  liver   18</span>
<span class="co">#&gt;  2: GSM1321183  liver   20</span>
<span class="co">#&gt;  3: GSM1321184  liver   22</span>
<span class="co">#&gt;  4: GSM1321185  liver   24</span>
<span class="co">#&gt;  5: GSM1321186  liver   26</span>
<span class="co">#&gt;  6: GSM1321187  liver   28</span>
<span class="co">#&gt;  7: GSM1321188  liver   30</span>
<span class="co">#&gt;  8: GSM1321189  liver   32</span>
<span class="co">#&gt;  9: GSM1321190  liver   34</span>
<span class="co">#&gt; 10: GSM1321191  liver   36</span>
<span class="co">#&gt; 11: GSM1321192  liver   38</span>
<span class="co">#&gt; 12: GSM1321193  liver   40</span>
<span class="co">#&gt; 13: GSM1321194  liver   42</span>
<span class="co">#&gt; 14: GSM1321195  liver   44</span>
<span class="co">#&gt; 15: GSM1321196  liver   46</span>
<span class="co">#&gt; 16: GSM1321197  liver   48</span>
<span class="co">#&gt; 17: GSM1321198  liver   50</span>
<span class="co">#&gt; 18: GSM1321199  liver   52</span>
<span class="co">#&gt; 19: GSM1321200  liver   54</span>
<span class="co">#&gt; 20: GSM1321201  liver   56</span>
<span class="co">#&gt; 21: GSM1321202  liver   58</span>
<span class="co">#&gt; 22: GSM1321203  liver   60</span>
<span class="co">#&gt; 23: GSM1321204  liver   62</span>
<span class="co">#&gt; 24: GSM1321205  liver   64</span>
<span class="co">#&gt;         sample tissue time</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-linear-models">Fit linear models<a class="anchor" aria-label="anchor" href="#fit-linear-models"></a>
</h2>
<p>The first step is to fit a series of linear models based on periodic splines for each genomic feature, in this case each gene, using <a href="https://doi.org/doi:10.18129/B9.bioc.limma" class="external-link">limma</a>. <code><a href="../reference/getModelFit.html">getModelFit()</a></code> takes several arguments besides the expression data and metadata, but here we just use the defaults. For example, the data are from one condition, so we leave <code>condColname</code> as <code>NULL</code>. Also, the expression data are from microarrays and already log-transformed, so we leave <code>method</code> as <code>'trend'</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="../reference/getModelFit.html">getModelFit</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">metadata</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-posterior-fits">Get posterior fits<a class="anchor" aria-label="anchor" href="#get-posterior-fits"></a>
</h2>
<p>The next step is obtain posterior estimates of the model coefficients using <a href="https://doi.org/10.1038/s41588-018-0268-8" class="external-link">multivariate adaptive shrinkage</a> (<a href="https://stephenslab.github.io/mashr/index.html" class="external-link">mashr</a>), which learns patterns in the data and accounts for noise in the original fits.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPosteriorFit.html">getPosteriorFit</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-rhythm-statistics">Get rhythm statistics<a class="anchor" aria-label="anchor" href="#get-rhythm-statistics"></a>
</h2>
<p>We can now use the posterior fits to compute rhythm statistics, i.e.Â properties of the curve, for each gene.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rhyStats</span> <span class="op">=</span> <span class="fu"><a href="../reference/getRhythmStats.html">getRhythmStats</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
<p>We can examine the distributions of the statistics in various ways, such as ranking genes by peak-to-trough amplitude (no p-values necessary) or plotting peak-to-trough amplitude vs.Â peak phase.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rhyStats</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">peak_trough_amp</span><span class="op">)</span><span class="op">]</span>, nrows <span class="op">=</span> <span class="fl">10L</span><span class="op">)</span>
<span class="co">#&gt;       feature peak_phase peak_value trough_phase trough_value peak_trough_amp</span>
<span class="co">#&gt;  1:     13170  8.1935432  10.402785    23.083237     6.646895    3.7558902049</span>
<span class="co">#&gt;  2:     13088 17.3125618  10.537264     9.237815     8.674852    1.8624115294</span>
<span class="co">#&gt;  3:    266645 18.3447383   9.775430     8.906014     8.153673    1.6217570599</span>
<span class="co">#&gt;  4:     13869  8.7168031   8.585007    18.044704     7.192154    1.3928531097</span>
<span class="co">#&gt;  5:     68396 19.2644734   9.639321     8.736966     8.771817    0.8675034661</span>
<span class="co">#&gt; ---                                                                          </span>
<span class="co">#&gt; 46:     14616 23.6961643   5.422345     6.683518     5.418776    0.0035688775</span>
<span class="co">#&gt; 47:     18994 14.2784936   5.601345    22.177204     5.599508    0.0018374833</span>
<span class="co">#&gt; 48: 100009600  2.1265901   5.126086     9.417707     5.124330    0.0017555339</span>
<span class="co">#&gt; 49:     12613  0.6076071   5.049669     8.810137     5.048225    0.0014443188</span>
<span class="co">#&gt; 50: 115485688 19.7468356   4.235779     1.518992     4.235585    0.0001938504</span>
<span class="co">#&gt;        mesor</span>
<span class="co">#&gt;  1: 8.478881</span>
<span class="co">#&gt;  2: 9.761782</span>
<span class="co">#&gt;  3: 9.092473</span>
<span class="co">#&gt;  4: 7.853253</span>
<span class="co">#&gt;  5: 9.283822</span>
<span class="co">#&gt; ---         </span>
<span class="co">#&gt; 46: 5.420228</span>
<span class="co">#&gt; 47: 5.600595</span>
<span class="co">#&gt; 48: 5.125045</span>
<span class="co">#&gt; 49: 5.048962</span>
<span class="co">#&gt; 50: 4.235683</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">rhyStats</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">peak_phase</span>, y <span class="op">=</span> <span class="va">peak_trough_amp</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'Peak phase (h)'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Peak-to-trough amplitude (norm.)'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">24</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="rhythmicity_files/figure-html/plot_stats-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="get-observed-and-fitted-time-courses">Get observed and fitted time-courses<a class="anchor" aria-label="anchor" href="#get-observed-and-fitted-time-courses"></a>
</h2>
<p>We can also compute the expected measurements for one or more genes at one or more time-points, which correspond to the fitted curves. Here we plot the posterior fits and observed expression for three of the top rhythmic genes (converting from gene id to gene symbol).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span> <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span>
  id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'13088'</span>, <span class="st">'13170'</span>, <span class="st">'13869'</span><span class="op">)</span>,
  symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Cyp2b10'</span>, <span class="st">'Dbp'</span>, <span class="st">'Erbb4'</span><span class="op">)</span><span class="op">)</span>

<span class="va">measFit</span> <span class="op">=</span> <span class="fu"><a href="../reference/getExpectedMeas.html">getExpectedMeas</a></span><span class="op">(</span><span class="va">fit</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">24</span>, <span class="fl">0.5</span><span class="op">)</span>, features <span class="op">=</span> <span class="va">genes</span><span class="op">$</span><span class="va">id</span><span class="op">)</span>
<span class="va">measFit</span><span class="op">[</span><span class="va">genes</span>, <span class="va">symbol</span> <span class="op">:=</span> <span class="va">i.symbol</span>, on <span class="op">=</span> <span class="fu">.</span><span class="op">(</span>feature <span class="op">=</span> <span class="va">id</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">measFit</span>, nrows <span class="op">=</span> <span class="fl">10L</span><span class="op">)</span>
<span class="co">#&gt;      time feature     value  symbol</span>
<span class="co">#&gt;   1:  0.0   13088 10.048449 Cyp2b10</span>
<span class="co">#&gt;   2:  0.0   13170  6.726346     Dbp</span>
<span class="co">#&gt;   3:  0.0   13869  7.655222   Erbb4</span>
<span class="co">#&gt;   4:  0.5   13088 10.057168 Cyp2b10</span>
<span class="co">#&gt;   5:  0.5   13170  6.841706     Dbp</span>
<span class="co">#&gt;  ---                               </span>
<span class="co">#&gt; 143: 23.5   13170  6.662765     Dbp</span>
<span class="co">#&gt; 144: 23.5   13869  7.611846   Erbb4</span>
<span class="co">#&gt; 145: 24.0   13088 10.048449 Cyp2b10</span>
<span class="co">#&gt; 146: 24.0   13170  6.726346     Dbp</span>
<span class="co">#&gt; 147: 24.0   13869  7.655222   Erbb4</span></code></pre></div>
<p>Next we combine the observed expression data and metadata. The curves show how <code>limorhyde2</code> is able to fit non-sinusoidal rhythms.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">measObs</span> <span class="op">=</span> <span class="fu"><a href="../reference/mergeMeasMeta.html">mergeMeasMeta</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">metadata</span>, features <span class="op">=</span> <span class="va">genes</span><span class="op">$</span><span class="va">id</span><span class="op">)</span>
<span class="va">measObs</span><span class="op">[</span><span class="va">genes</span>, <span class="va">symbol</span> <span class="op">:=</span> <span class="va">i.symbol</span>, on <span class="op">=</span> <span class="fu">.</span><span class="op">(</span>feature <span class="op">=</span> <span class="va">id</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">measObs</span>, nrows <span class="op">=</span> <span class="fl">10L</span><span class="op">)</span>
<span class="co">#&gt;         sample tissue time feature      meas  symbol</span>
<span class="co">#&gt;  1: GSM1321182  liver   18   13088 10.209968 Cyp2b10</span>
<span class="co">#&gt;  2: GSM1321182  liver   18   13170  7.871989     Dbp</span>
<span class="co">#&gt;  3: GSM1321182  liver   18   13869  6.851685   Erbb4</span>
<span class="co">#&gt;  4: GSM1321183  liver   20   13088 10.503358 Cyp2b10</span>
<span class="co">#&gt;  5: GSM1321183  liver   20   13170  7.281691     Dbp</span>
<span class="co">#&gt; ---                                                 </span>
<span class="co">#&gt; 68: GSM1321204  liver   62   13170  8.656560     Dbp</span>
<span class="co">#&gt; 69: GSM1321204  liver   62   13869  6.970524   Erbb4</span>
<span class="co">#&gt; 70: GSM1321205  liver   64   13088 10.352095 Cyp2b10</span>
<span class="co">#&gt; 71: GSM1321205  liver   64   13170  8.157525     Dbp</span>
<span class="co">#&gt; 72: GSM1321205  liver   64   13869  7.048457   Erbb4</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">'free_y'</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">time</span>, y <span class="op">=</span> <span class="va">value</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">measFit</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">time</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">24</span>, y <span class="op">=</span> <span class="va">meas</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fl">21</span>, size <span class="op">=</span> <span class="fl">1.5</span>,
             data <span class="op">=</span> <span class="va">measObs</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Circadian time (h)'</span>, y <span class="op">=</span> <span class="st">'Expression (norm.)'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">24</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="rhythmicity_files/figure-html/plot_timecourse-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jake Hughey, Dora Obodo, Elliot Outland.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
